support/util/expose.go

```diff
@@ -31,6 +46,21 @@ func UseDedicatedDNSforKAS(hcp *hyperv1.HostedControlPlane) bool {
 		apiServerService.Route != nil && apiServerService.Route.Hostname != ""
 }
 
+func UseDedicatedDNSForOAuth(hcp *hyperv1.HostedControlPlane) bool {
+	oauthService := ServicePublishingStrategyByTypeForHCP(hcp, hyperv1.OAuthServer)
+	return IsRouteOAuth(hcp) && oauthService.Route != nil && oauthService.Route.Hostname != ""
+}
+
+func UseDedicatedDNSForKonnectivity(hcp *hyperv1.HostedControlPlane) bool {
+	konnService := ServicePublishingStrategyByTypeForHCP(hcp, hyperv1.Konnectivity)
+	return IsRouteKonnectivity(hcp) && konnService.Route != nil && konnService.Route.Hostname != ""
+}
+
+func UseDedicatedDNSForIgnition(hcp *hyperv1.HostedControlPlane) bool {
+	ignitionService := ServicePublishingStrategyByTypeForHCP(hcp, hyperv1.Ignition)
+	return IsRouteIgnition(hcp) && ignitionService.Route != nil && ignitionService.Route.Hostname != ""
+}
+
```

Comment:
those functions could be merged into one function that takes the service type as an argument

Reply:
@muraee @bryan-cox Alright, pushed one more commit to merge those functions and remove some duplication.